readme.md

```diff
@@ -50,6 +50,7 @@ Practical examples are coming soon.
 
 `remark-language-server` uses the same configuration files as
 [`remark-cli`][remark-cli].
+Language clients should notify the language server if these files change.
```

Comment:
It may be good to make an exact list of these files, ideally by putting this in a dedicated section in https://github.com/remarkjs/remark/blob/main/packages/remark-cli/readme.md.

Reply:
it says that right above examples? https://github.com/remarkjs/remark/blob/main/packages/remark-cli/readme.md#examples Or what are you thinking off?

I still don’t get what I’d have to if I were to use this. “These files” is vague to me. Should particular *paths* be watched? Or basenames on an entire file system? Do I need to set up chokidar or something? unified-engine is going to walk “outside” of VSC workspaces. Should say `~/.remarkrc` also be watched? Is it clear how to notify this language server of changes?

Reply:
> it says that right above examples? https://github.com/remarkjs/remark/blob/main/packages/remark-cli/readme.md#examples Or what are you thinking off?

I think this demonstrates the problem. You link a section, then tell me to scroll up. It would be nice if we could link to https://github.com/remarkjs/remark/blob/main/packages/remark-cli/readme.md#configuration instead.

Above https://github.com/remarkjs/remark/blob/main/packages/remark-cli/readme.md#compatibility is a full list of configuration files. I think it’s good to move this full list into the _Configuration_ section I just proposed. It’s an important part of `remark-cli` for end users to unserstand, so I think if this information is easily accessible without any form of redirection to an example of `unified-engine`.

This also makes me think it’s probably best to duplicate the list here in `remark-language-server`. It’s not likely to change anyway.

> Do I need to set up chokidar or something? […] Is it clear how to notify this language server of changes?

The language client needs to setup a file watcher. Whatever this means depends on the language client library. How this works should generally be clear for people working with their library.

> unified-engine is going to walk “outside” of VSC workspaces. Should say `~/.remarkrc` also be watched?

This is a good question. I will mention this, but leave the decision up to the implementator.

Reply:
I mean, it’s in the docs, I tried to link as close as can be 🤷‍♂️

The downside w adding a whole section on how config files works in remark-cli is that we had that. And we have similar “problems” elsewhere. It means duplicating content from lower-level projects upwards.
I’m always open to improving this, but that’s an important trade off. Should things be copy/pasted, or short in higher projects and explained in detail in the projects that actually do the work?

As for this OG comment, feel free to PR to improve the `remark-cli` readme!

Reply:
I think this list is now fine as-is. The most important goal here is to list which files need to be watched for projects that implement the `remark-language-server`.